pkg/license/reporter_test.go

```diff
@@ -0,0 +1,162 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package license
+
+import (
+	"testing"
+
+	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
+	kbv1 "github.com/elastic/cloud-on-k8s/pkg/apis/kibana/v1"
+	essettings "github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/settings"
+	kbconfig "github.com/elastic/cloud-on-k8s/pkg/controller/kibana/config"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func Test_Get(t *testing.T) {
```

Comment:
This is the "integration test" I was looking for in my previous comment: seems god enough for me to keep it as a unit test here 👍 

It would be nice if we could add another one to test the configmap reporting?
We could have ResourceReporter.Start accept the `refreshPeriod ` as a parameter, call it in the test with a super small `refreshPeriod`, and have the test check:
- a configmap is eventually created with the right content
- modify the ES spec
- the configmap should eventually be updated with the right content

Reply:
+100